airbyte-integrations/connectors/source-hubspot/source_hubspot/streams.py

```diff
@@ -838,21 +843,19 @@ def read_records(
 
             for record in records:
                 cursor = self._field_to_datetime(record[self.updated_at_field])
-                latest_cursor = max(cursor, latest_cursor) if latest_cursor else cursor
+                self._update_state(latest_cursor=cursor)
```

Comment:
This change introduces a critical issue on our end.

Because the state is updated at each request, as soon as all the 100 objects in the response have been updated (last_modified_at) at the exact same time (by a Hubspot import, a batch update, etc), it creates an infinite loop.
![Screenshot 2022-07-24 at 22 49 24](https://user-images.githubusercontent.com/65104164/180665340-c492ffe1-6ca8-4eee-8a92-827ed6ee859b.png)

The state should be updated only after the 10k batch as before.

It's also very verbose in the logs.
![Screenshot 2022-07-24 at 22 47 30](https://user-images.githubusercontent.com/65104164/180665266-bd899c42-f97b-41cd-86dd-25788272ccf4.png)


We have to roll back on the previous version (0.1.74).

Reply:
@davydov-d @drrest WDTY?

Reply:
@elliottrabac you're right, I'm gonna revert the whole PR

Reply:
@elliottrabac take a look please https://github.com/airbytehq/airbyte/pull/14999

Reply:
@elliottrabac this PR has been rolled back, please try again